<?xml version="1.0"?>
<robot xmlns:xacro="http://ros.org/wiki/xacro" name="igus_4DOF_SV">

<!-- URDF file for the Commonplace Robotics Mover4 robot arm -->
<!-- Version 1.1 from Oct. 04th, 2016. -->
  
  <link name="base_link">
    <visual>
	  <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
          <mesh filename="package://cpr_robot/robots/igus_4DOF_SV/Joint0.dae"  scale="0.001 0.001 0.001"/>
        </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
          <mesh filename="package://cpr_robot/robots/igus_4DOF_SV/Joint0Coll.dae" scale="0.001 0.001 0.001"/>
        </geometry>
    </collision>
  </link>
  <link name="link1">
    <visual>
	  <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
          <mesh filename="package://cpr_robot/robots/igus_4DOF_SV/Joint1.dae"  scale="0.001 0.001 0.001"/>
        </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
          <mesh filename="package://cpr_robot/robots/igus_4DOF_SV/Joint1Coll.dae" scale="0.001 0.001 0.001"/>
        </geometry>
    </collision>
  </link>
  <link name="link2">
    <visual>
	  <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
          <mesh filename="package://cpr_robot/robots/igus_4DOF_SV/Joint2.dae"  scale="0.001 0.001 0.001"/>
        </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
          <mesh filename="package://cpr_robot/robots/igus_4DOF_SV/Joint2Coll.dae" scale="0.001 0.001 0.001"/>
        </geometry>
    </collision>
  </link>
  <link name="link3">
    <visual>
	  <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
          <mesh filename="package://cpr_robot/robots/igus_4DOF_SV/Joint3.dae"  scale="0.001 0.001 0.001"/>
        </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
          <mesh filename="package://cpr_robot/robots/igus_4DOF_SV/Joint3Coll.dae" scale="0.001 0.001 0.001"/>
        </geometry>
    </collision>
  </link>
  <link name="link4">
    <visual>
	  <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
          <mesh filename="package://cpr_robot/robots/igus_4DOF_SV/Joint4.dae"  scale="0.001 0.001 0.001"/>
        </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
          <mesh filename="package://cpr_robot/robots/igus_4DOF_SV/Joint4.dae" scale="0.001 0.001 0.001"/>
        </geometry>
    </collision>
  </link>
  <joint name="joint1" type="continuous">
    <axis xyz="0 0 1"/>
    <parent link="base_link"/>
    <child link="link1"/>
    <origin rpy="0 0 0" xyz="0 0 0.1275"/>
    <limit effort="100" velocity="30"/>
    <joint_properties damping="0.0" friction="0.0"/>
  </joint>
  <joint name="joint2" type="continuous">
    <axis xyz="0 1 0"/>
    <parent link="link1"/>
    <child link="link2"/>
    <origin rpy="0 0 0" xyz="0 0 0.10025"/>
    <limit effort="100" velocity="30"/>
    <joint_properties damping="0.0" friction="0.0"/>
  </joint>
 <joint name="joint3" type="continuous">
    <axis xyz="0 1 0"/>
    <parent link="link2"/>
    <child link="link3"/>
    <origin rpy="0 1.57079632679 0" xyz="0 0 0.270"/>
    <limit effort="100" velocity="30"/>
    <joint_properties damping="0.0" friction="0.0"/>
  </joint>
 <joint name="joint4" type="continuous">
    <axis xyz="0 1 0"/>
    <parent link="link3"/>
    <child link="link4"/>
    <origin rpy="0 0 0" xyz="0 0 0.240"/>
    <limit effort="100" velocity="30"/>
    <joint_properties damping="0.0" friction="0.0"/>
  </joint>
</robot>

